### ISSUE TITLE
Prevent multiple gpodder instances

### ISSUE THREAD

**Issue by auouymous** [gpodder/gpodder#1351]
On Linux, running a second instance switches focus to the first instance. This prevents a blank gpodder window and hidden `sqlite3.OperationalError: database is locked` messages in the logs. The fake dbus implementation on Windows does not provide this feature and results in confusion. #1253 would add a real dbus implementation to Windows and resolve this issue, so dbus/MPRIS support in media programs on Windows no longer matters. I don't know if Mac has a working dbus or if it uses the fake dbus as well.

But is switching to the other instance the best choice? Shortly before the 3.11.0 release, gpodder's window was closing but the process wasn't terminating. If gpodder was in such a state, the second instance won't open, leading to confusion.

Would it be better to detect a locked database, open a warning dialog and then quit?

**Comment by anewuser** [comment]
Isn't it possible to check for a process *and* a window?

1. Check if there's any gPodder process running
no -> start gPodder
yes -> go to 2

2. Is the gPodder window open?
no -> show a warning and then quit
yes -> bring the window to the front

**Comment by auouymous** [comment]
It can query a list of windows from X11 but I don't know if there is a portable way to do that across Mac and Windows.

The second instance sends a signal over dbus telling the program to show its window. That signal probably won't be run if its process is blocked somewhere else. So we could have the signal return a value, and the second instance could wait a few seconds before timing out and showing a dialog to kill other instances.

It could then perform the database lock check after the dbus check and show a similar dialog.

**Comment by anewuser** [comment]
If this means spending too much time working on this, I agree with your original comment:

>Would it be better to detect a locked database, open a warning dialog and then quit?

After displaying the warning dialog, you can expect the user to find the window (or ghost process) by himself/herself. The warning can include a note like "Please try restarting your computer if this problem persists." for users unable to terminate it.

**Comment by auouymous** [comment]
Good idea. Restarting the computer is often an easier solution for many people.
